### PR TITLE
[Proposal] OpenShift-Ansible Proposal Process

### PR DESCRIPTION
**Pull request by mtnbikenc**
As discussed on the OpenShift-Ansible Architecture Meeting, this is the proposed process for proposals.

This is based on http://commissaire.readthedocs.io/en/latest/cpd/1.html

### CODE REVIEW

**Comment by ashcrow** [comment]
I like it, but I'm bias 😆. The only thing I can think of adding which may be helpful here would possibly be a time limit so old proposals "time out" if no one (including the proposer) leaves the proposal with no updates.

File: docs/proposals/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  * If 75% or more of the active development team give the proposal a :+1: it is Approved
+  * If 50% or more of the active development team disagrees with the proposal it is Closed
+  * If the person proposing the proposal no longer wishes to continue they can request it to be Closed
+  * If none of the above is met the cycle can continue to Step 4.
+6. The current development lead(s) pull in the proposal to the docs folder and update the status
+7. The current development lead(s) update the issue with the result
+
+
+Once a proposal is approved, a Trello card should be created and scheduled for
+implementation.  Stale proposals will be closed after $some_time.
```
Comment (mtnbikenc) [inline docs/proposals/README.md]: @ashcrow I was attempting to include a time limit here but didn't know what to put for `$some_time`.  How long after no updates is too long before it is stale?  One month? How long should we give for feedback?  Two-weeks?

Comment (ashcrow) [reply docs/proposals/README.md]: I'd say if no comments or updates are made in a 2 week period the proposal should closed.

**Comment by ingvagabund** [comment]
As it has been proved many times all proposals have tendency to rot unless a minimum activity from a proposal maker and the audience is generated. Every proposal should have another person who will be in the war zone together with the proposal maker and support his ideas, his justification. As long as we will go with 1 proposal 1 person, it will take a lot of time to shape all the relevant ideas. What about to have two persons that will work on the proposal together (the role of maker and defender can switch over the time between folks). E.g. I will be working on the proposal of action based roles and @michaelgugino will cover my back, will be a friend I will talk to when discussing parts of my proposal so I will get a feedback asap, I will get different opinion before I schedule a new 10 minutes movie on the arch meeting.

**Comment by ingvagabund** [comment]
I am not saying there is someone who can't make a proposal on its own. Just saying we should support each other more often, e.g. arguments, reviews, @tbielawa 's back when they attack again :P, ...

**Comment by mtnbikenc** [comment]
@abutcher @ewolinetz @giuseppe @ingvagabund @kwoodson @michaelgugino @sdodson @ashcrow @tbielawa 

I've added each of you to the Reviewers list.  Please either Approve or Request Changes so we can continue to move this forward.  Thanks!

**Comment by mtnbikenc** [comment]
@ingvagabund The person making the proposal could reach out and find someone to work together and have that person added to the Assignees list.  That could indicate who is working together on getting the proposal through.

**Comment by ashcrow** [comment]
@ingvagabund I'd be OK with having two people as a requirement for ownership as part of a proposal ... though I could see some proposals being done by a single person while others might be done by X number of people. 🤷‍♂️

**Comment by ingvagabund** [comment]
> In the meantime each proposal could define it's own `$sometime` until we figure out what works.

We could set the deadline in a trello card.

File: docs/proposals/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# OpenShift-Ansible Proposal Process
+
+The following process should be followed when a proposal is needed:
+
+1. Create a pull request with the initial proposal using the proposal template
```
Comment (michaelgugino) [inline docs/proposals/README.md]: Where does the proposal live in tree?  How should it be named?  This is probably obviously, but we might as well spell it out.

**Comment by mtnbikenc** [comment]
@michaelgugino I agree with your comments and see those suggestions as great items for follow-up updates to the proposal process.  Is there anything here you don't agree with and would like to Request Changes?

**Comment by michaelgugino** [comment]
@mtnbikenc The only addition I see as necessary to this PR is noting path and naming convention of proposals for clarity's sake.

File: docs/proposals/README.md (hunk 1 above)
Comment (sdodson) [reply docs/proposals/README.md]: This is probably a bit aggressive given how overloaded most of us are but we can adjust in the future.

**Comment by mtnbikenc** [comment]
@michaelgugino Something like this?

```
1. Create a pull request with the initial proposal
  * Use the [proposal template][template]
  * Name the proposal using two or three topic words with underscores as a separator (i.e. proposal_template.md)
  * Place the proposal in the docs/proposals directory
```

**Comment by michaelgugino** [comment]
@mtnbikenc that looks good to me.

**Comment by mtnbikenc** [comment]
Pushed an update with details requested by @michaelgugino.

**Comment by mtnbikenc** [comment]
@abutcher @ewolinetz @giuseppe @ingvagabund @kwoodson @michaelgugino @sdodson  @tbielawa 

Thanks for the prior reviews.  I believe I've addressed any concerns in comments, however, the latest commit has cleared previous reviews.  Could you please review again?

File: docs/proposals/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+TODO: Add details about when a proposal is or is not required. 
+
+## Proposal Process
+The following process should be followed when a proposal is needed:
+
+1. Create a pull request with the initial proposal
+  * Use the [proposal template][template]
+  * Name the proposal using two or three topic words with underscores as a separator (i.e. proposal_template.md)
+  * Place the proposal in the docs/proposals directory
+2. Notify the development team of the proposal and request feedback
```
Comment (tbielawa) [inline docs/proposals/README.md]: Please define 'development team' and is there a way that word can link to some kind of github generated list of said team members?

File: docs/proposals/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+3. Review the proposal on the OpenShift-Ansible Architecture Meeting
```
Comment (tbielawa) [inline docs/proposals/README.md]: Link to mtg doc?

File: docs/proposals/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+5. Approved/Closed Phase
+  * If 75% or more of the active development team give the proposal a :+1: it is Approved
+  * If 50% or more of the active development team disagrees with the proposal it is Closed
+  * If the person proposing the proposal no longer wishes to continue they can request it to be Closed
+  * If there is no activity on a proposal, the active development team may Close the proposal at their discretion
+  * If none of the above is met the cycle can continue to Step 4.
+6. For approved proposals, the current development lead(s) will:
+  * Update the Pull Request with the result and merge the proposal
+  * Create a card on the Cluster Lifecycle [Trello board][trello] so it may be scheduled for implementation.
+
```
Comment (tbielawa) [inline docs/proposals/README.md]: Procedure for bringing a failed/closed proposal back into review later on?

**Pull request merged**